### PR TITLE
Guided setup

### PR DESCRIPTION
**Pull request by joseivanlopez**
PBI: https://trello.com/c/crhdffWC/516-5-storageng-guided-setup

This PR replaces this old one: https://github.com/yast/yast-storage-ng/pull/193

### CODE REVIEW

File: src/lib/y2storage/dialogs/guided_setup/select_scheme.rb [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    class GuidedSetup
+      # Dialog to select partitioning scheme.
+      class SelectScheme < Base
+        def encryption_handler
+          widget_update(:password, widget_value(:encryption), attr: :Enabled)
+          widget_update(:repeat_password, widget_value(:encryption), attr: :Enabled)
+        end
+
+        def back_handler
+          unload_cracklib
```
Comment (ancorgs) [inline src/lib/y2storage/dialogs/guided_setup/select_scheme.rb]: Wouldn't it be better to redefine `#finish_dialog` or `#close_dialog` in order to unload cracklib? It sounds more robust to me than doing it in every possible handler.

File: src/lib/y2storage/filesystems/type.rb [hunk 2]
```diff
@@ -30,6 +30,82 @@ class Type
       include StorageEnumWrapper
 
       wrap_enum "FsType"
+
+      FILESYSTEMS = {
```
Comment (ancorgs) [inline src/lib/y2storage/filesystems/type.rb]: nitpicking: not sure if I like the name. What about `PROPERTIES` or something similar?

File: src/lib/y2storage/filesystems/type.rb [hunk 3]
```diff
@@ -30,6 +30,82 @@ class Type
...
+          name: "ISO9660"
+        },
+        udf:      {
+          name: "UDF"
+        }
+      }
+
+      ROOT_FILESYSTEMS = [:ext2, :ext3, :ext4, :btrfs, :reiserfs, :xfs]
+
+      HOME_FILESYSTEMS = [:ext2, :ext3, :ext4, :btrfs, :reiserfs, :xfs]
```
Comment (ancorgs) [inline src/lib/y2storage/filesystems/type.rb]: I would use `private_constant` in all these constants.

File: src/lib/y2storage/filesystems/type.rb [hunk 4]
```diff
@@ -30,6 +30,82 @@ class Type
...
+      # @return [Array<Filesystems::Type>]
+      def self.home_filesystems
+        HOME_FILESYSTEMS.map { |f| find(f) }
+      end
+
+      # Human readable text for a filesystem
+      #
+      # @return [String]
+      def to_human
+        FILESYSTEMS[to_sym][:name]
```
Comment (ancorgs) [inline src/lib/y2storage/filesystems/type.rb]: Doesn't look robust. What if a new type is added in libstorage but we don't add the corresponding name?

File: src/lib/y2storage/dialogs/guided_setup/select_scheme.rb (hunk 1 above)
Comment (joseivanlopez) [inline src/lib/y2storage/dialogs/guided_setup/select_scheme.rb]: Done

File: src/lib/y2storage/filesystems/type.rb (hunk 2 above)
Comment (joseivanlopez) [inline src/lib/y2storage/filesystems/type.rb]: Yes, maybe PROPERTIES is a better name for that.

File: src/lib/y2storage/filesystems/type.rb (hunk 3 above)
Comment (joseivanlopez) [inline src/lib/y2storage/filesystems/type.rb]: Ok

File: src/lib/y2storage/filesystems/type.rb (hunk 4 above)
Comment (joseivanlopez) [inline src/lib/y2storage/filesystems/type.rb]: Yes, you are right.

**Pull request merged**